pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -68,6 +68,32 @@ func (c *Reconciler) createAffinityAssistants(ctx context.Context, wb []v1beta1.
 				if err == nil {
 					logger.Infof("Created StatefulSet %s in namespace %s", affinityAssistantName, namespace)
 				}
+			case err == nil && a != nil && a.Status.ReadyReplicas == 1:
+				// the node which hosts the affinity assistant pod is unschedulable or cordoned for some reason,
+				// delete the affinity assistant pod such that a StatefulSet can recreate the same pod on a different node
+				// get the list of pods associated with a given StatefulSet based on the following label:
+				// app.kubernetes.io/instance=affinity-assistant-641c8a0038
+				pods, err := c.KubeClientSet.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+					LabelSelector: workspace.LabelInstance + "=" + a.Name,
+				})
+				if err != nil || len(pods.Items) == 0 {
```

Comment:
I'm not sure why len(pods.Items) would be 0 here, but would it make sense to allow this to continue in that case, or no?

Reply:
The `err` represents an error from `Get` i.e. retrieving a list of pods caused any failure or not. It could very well return an empty list if the necessary label is not found. i.e. the required pod was found. Such `pod not found` is not valid must be reported.